manifests/init.pp

```diff
@@ -29,8 +30,10 @@
   }
 
   file { $confdir:
-    ensure => directory,
-    mode   => '0755',
+    ensure  => directory,
+    mode    => '0755',
+    recurse => $purge_confdir,
```

Comment:
I'm not liking the ability for `recurse => true` here.  Would it still work for you if only `purge` was set?